### ISSUE TITLE
problem at update from 7.0.4-17.1 to 8.0.0-0.5.1.RC1

### ISSUE THREAD

**Issue by michl080** [owncloud/core#13917]
good morning. I tried to update my installation from V7 to V8 and ran into a problem. 

Environment: Centos 6 fully patched. 
used repo: http://download.opensuse.org/repositories/isv:/ownCloud:/community:/8.0/CentOS_6/
used command to update from V7 to V8: yum update (with activated repository above)
error message in /var/log/http/ssl_error_log : 
`PHP Parse error:  syntax error, unexpected '[' in /var/www/html/owncloud/lib/base.php on line 849`
symptom: blank page displayed instead of login page.
No error in owncloud.log displayed.
No changes in setup before or after update.

I hope this is enough information.

**Comment by MorrisJobke** [comment]
For owncloud 8 you need at least PHP 5.4. It doesn't seem that you have this installed.

**Comment by LukasReschke** [comment]
Warning will added with https://github.com/owncloud/core/pull/13918

**Comment by LukasReschke** [comment]
@jnweiger Can we somehow require at least 5.4 for 8.0?

**Comment by MorrisJobke** [comment]
The package seems to be named `php54w`

**Comment by DeepDiver1975** [comment]
> @jnweiger Can we somehow require at least 5.4 for 8.0?

no idea if the explicit version of php is exposed in the rpm meta data

**Comment by MorrisJobke** [comment]
https://webtatic.com/packages/php54/ <- I just found that *hides*

**Comment by jnweiger** [comment]
@DeepDiver1975 can you confirm that owncloud 8 requires php5.4?
if so, I'll add it as a versioned requires! This is important. 

All the centos6 builds are useless then. OBS does not offer any php53 for centos6.

**Comment by RealRancor** [comment]
@jnweiger 

Yes, PHP 5.4 is required for OC8:

http://doc.owncloud.org/server/8.0/admin_manual/whats_new_admin.html#no-more-php-5-3
https://owncloud.org/blog/what-is-coming-in-owncloud-server-8-0-for-system-administrators/

**Comment by jnweiger** [comment]
Thanks, @RealRancor adding a hard dependency on the linux packages now. 

We will no longer have packages for SLES-11-SP3, Debian-6, CentOS-6.

**Comment by jnweiger** [comment]
Rebuild 8.0.0~RC1-5 will have hard php >= 5.4.0 requirements.

**Comment by MorrisJobke** [comment]
> We will no longer have packages for SLES-11-SP3, Debian-6, CentOS-6.

Why?

**Comment by DeepDiver1975** [comment]
> Why?

because they provide php 5.3 only

**Comment by MorrisJobke** [comment]
Ah :) Got it :)

**Comment by jnfrmarks** [comment]
@MTRichards 

Adding you into this conversation around no longer supporting some packages.

**Comment by jnweiger** [comment]
php54 support for CentOS6 comes from e.g. webtatic, epel, scl, or remi repositories. We have none of them in the OBS.  A supported php54 for Debian-6 or SLE-11 does not exist afaik.
Hence my concern that we cannot provide one-click solutions there.

**Comment by MTRichards** [comment]
I agree that a single click is not possible for these distros. Clearly we will support ownCloud on those distros, but only if they install PHP 5.4 through their selected OS. 

Why? 
PHP 5.4 has been EoL for months now. Any problems in PHP 5.3 affecting ownCloud - and there are many - will be blamed on oC, and we can't fix them - http://php.net/supported-versions.php for more on the support lifecycle of these versions.

We will have the same problem with PHP 5.4 at the end of this year.

For now, @jnweiger is there a way to remove the PHP dependency for CentOS 6, SLES 11.3 and Debian 6 and still have a package of ownCloud installed - without the PHP check? We may have to provide notes in the documentation about this so people don't get lost, and then have to manually install the PHP packages.

We also need to document that upgrading on these OSes will be a problem unless PHP 5.4 is installed. 
@carlaschroder needs documentation and release note.

**Comment by jnweiger** [comment]
@MTRichards Until this morning we had ownCloud packages for CentOS 6, SLES 11.3 and Debian 6  without a strict version check. 

The comments here seem to agree that we should not offer packages that would not work.
Your comment disagrees with that. 

As of now, my packages will refuse to install on a system where php 5.3 is present.
I an easily revert to the previous situation, where I only requested 5.3.3 as a minimum.

**Comment by MTRichards** [comment]
Oh no, please don't revert. I seek first to understand and then maybe make changes.

Refuse to install on a system with 5.3=good. So one click install doesn't work, but if someone goes out and adds PHP 5.4, then it will install - even on CentOS 6, SLES 11 SP3, etc <-correct?

If so, don't change anything.

**Comment by RealRancor** [comment]
> For now, @jnweiger is there a way to remove the PHP dependency for CentOS 6, SLES 11.3 and Debian 6 and still have a package of ownCloud installed - without the PHP check? We may have to provide notes in the documentation about this so people don't get lost, and then have to manually install the PHP packages.

That will be funny. From my experiences most users just don't check the documentation.

**Comment by jnweiger** [comment]
@MTRichards thanks for the clarification. Yes, if someone adds PHP 5.4 then installation will succeed.

I agree with @RealRancor that users will normally skip the docs. So a very clear message at runtime must be show, in case users trick the installer and try force install on systems with PHP 5.3 (I am afraid, this is more likely than docs being read...)

**Comment by michl080** [comment]
Hi everybody,
I followed your discussion and updated to PHP5.4 using the centos scl repository. Now trying to remove the old and obsoloete PHP5.3 throws this error message:
`Removing:
 php                                                             x86_64                                  5.3.3-40.el6_6                                    @updates                                                     3.5 M`
`Removing for dependencies:
 owncloud                                                        noarch                                  8.0.0-0.7.1.RC1                                   @isv_ownCloud_community_8.0                                   32 k
`
and all components of owncloud

This is really ugly. I cannot use PHP5.3, but I cannot unistall it either!

**Comment by jnweiger** [comment]
@michl080 can you give a step by step instruction how to use centos scl? I was only aware of redhat scl, which would *not* uninstall php-5.3.3

**Comment by jnweiger** [comment]
@michl080 This specific issue will go away with the next rebuild. Then you will be forced to upgrade php before you can install owncloud-8.0.0

**Comment by michl080** [comment]
@jnweiger  You are right, Centos scl will not unistall php-5.3 while installing php-5.4.

Ok, see the initial steps in the first posting. 
From there I installed the SCL repo with
`yum install centos-release-SCL` 
and installed all existing PHP5.3 packages in their equivalent PHP5.4 version using 
`yum install php54 php54-php-pdo php54-php-mbstring php54-php-intl php54-php-mysql php54-php-process php54-php-gd`
Next step was to remove the old php5.3 version using 
`yum remove php`.
 But this resulted in the above error message where the system tried to resolve the dependency with owncloud by trying to remove not only php5.3, but also owncloud.

It will be interesting to see how I can free myself out of my personal problem with the owncloud 0:8.0.0-0.7.1.RC1 build.

**Comment by SergioBertolinSG** [comment]
Please note that a full guide should be needed to upgrade php version in centos 6 first, after replacing php 5.3.3 with php 5.4 using webtatic procedure installing owncloud package has this problem (probably more) Error: php54w-common conflicts with php-common-5.3.3-40.el6_6.i686

**Comment by jnweiger** [comment]
I am stuck here:

    yum install centos-release-SCL 
    scl enable php54 'php --version'
     PHP 5.4.16 (cli) (built: Nov 19 2014 08:05:17) 
     ...
    scl enable php54 'yum install -y owncloud'
     Error: Package: owncloud-server-8.0.0-0.8.1.RC1.noarch (isv_ownCloud_community_8.0)
           Requires: php >= 5.4.0
           Available: php-5.3.3-38.el6.x86_64 (base)

**Comment by DeepDiver1975** [comment]
> scl enable php54 'php --version'

because the package is called php54 and not php ???

**Comment by michl080** [comment]
morning,
apparently, the concept of scl is that both the original php-5.3 and the SCL-version php-5.4 are available. The SCL version is selected through the "scl" command, but the "standard-version" is still 5.3.
see my solution below...

**Comment by michl080** [comment]
I could get the OC-8-RC2 to run on CENTOS6, but I had to go the "brute" way.

`yum remove php php-cli php-common php-gd php-intl php-mbstring php-mysql php-pdo
php-process php-xml`
This did not only remove all the packages above, but also Owncloud in the version that was installed at that time (8.0.0-0.7.1.RC1)

Please note that this was easy on my machine, because no other application relied on php. It may be more complicated on other machines.

Next step is to activate the `http://webtatic.com/packages/php55` repo.
Note that I decided to install not only php-5.4, but moved directly to php-5.5. 
`yum install php55w php55w-cli php55w-common php55w-gd php55w-intl
php55w-mbstring php55w-mysqlnd php55w-pdo php55w-process php55w-xml
`
this resulted in 
`#  php -v
PHP 5.5.21 (cli) (built: Jan 23 2015 01:30:54)
Copyright (c) 1997-2014 The PHP Group
Zend Engine v2.5.0, Copyright (c) 1998-2014 Zend Technologies`

good!

Next step was `yum install owncloud` which installed also automagically all apps correctly.

voila, that is it, my `owncloud.noarch 0:8.0.0-0.2.1.RC2` is up and running!

thanks for all you help and proposals!

**Comment by DeepDiver1975** [comment]
@jnweiger we can set an alternative dependency on php54 | php55 | php (>= 5.4)

in this case we support alterative repos on centos as well - same should work for debian - there is a repo https://www.dotdeb.org/ which provides higher php versions - but this is topic for another issue.

**Comment by jnweiger** [comment]
RPM does not allow to require 'php54 (>= 5.4.0)' OR 'php (>= 5.4.0)'. It only allows AND logic between Require statements. Evaluating a workaround with virtual provides.... sigh.

**Comment by DeepDiver1975** [comment]
> RPM does not allow to require 'php54 (>= 5.4.0)' OR 'php (>= 5.4.0)'. It only allows AND logic between Require statements. Evaluating a workaround with multiple subpackages and virtual provides.... sigh.

so let's skip this for now

**Comment by jnweiger** [comment]
Commited revision 71. project:isv:ownCloud:community:8.0
Should have a working owncloud-server-scl-php54 package that allows clean installation of owncloud 8.0.0~RC2 on centos6 with scl php54. The trick is: yum can choose between owncloud-server and owncloud-server-scl-php54 both providing 'owncloud-server'. One requires php >= 5.4.0 the other requires php54.

**Comment by jnweiger** [comment]
Another rebuild needed. The virtual provide must be different than the package name. Changed to 'owncloud-server-core'. Requiring php54 is not sufficient. We also need to require php54-php, this brings modphp for apache.

**Comment by jnweiger** [comment]
CentOS6 is back with commit#78 at isv:ownCloud:community:8.0/owncloud !

Install procedure for CentOS6 is this:

    yum install -y centos-release-SCL
    yum install -y php54 wget
    wget -nv http://download.opensuse.org/repositories/isv:/ownCloud:/community:/8.0/CentOS_6/isv:ownCloud:community:8.0.repo -O /etc/yum.repos.d/isv:ownCloud:community:8.0.repo
    yum install -y owncloud

**Comment by RealRancor** [comment]
@jnweiger 
Is it possible to add those instructions to the OBS description of CenOS6 on:

http://software.opensuse.org/download/package?project=isv:ownCloud:community&package=owncloud

or are those instructions auto-generated?

**Comment by tuxmaster** [comment]
Tested the last post, but it will fails with:
Error: Package: owncloud-server-8.0.0-9.1.noarch (isv_ownCloud_community_8.0)
           Requires: php >= 5.4.0
           Installed: php-5.3.3-40.el6_6.x86_64 (@updates)
               php = 5.3.3-40.el6_6
           Available: php-5.3.3-38.el6.x86_64 (base)
               php = 5.3.3-38.el6
I think the problem is, that the php package from SCL is called php54 instant php.
rpm -q php54:
php54-1.1-5.el6.centos.alt.x86_64

**Comment by tuxmaster** [comment]
yum install owncloud-server-scl-php54 due the job :)

**Comment by koumdros** [comment]
ifollowed the isntruction above,after installing php54 packages via scl repo, then used the 
"yum install owncloud-server-scl-php54" command , which did download the new version of owncloud, 

after reloading apache i got a first page telling em that OC 8 would not run as i had php 5.3 and not 5.4.
I then disabled the php 5.3.3.0 module in /etc/httpd/conf.d directory, adn reloaded apache.

now the first page of Owncloud seemingly appears but i get an error like this:

Internal Server Error
The server encountered an internal error and was unable to complete your request.
Please contact the server administrator if this error reappears multiple times, please include the technical details below in your report.
More details can be found in the server log.

checking the server log I got this:
```
#0 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Connection.php(816): OC\\\\DB\\\\Connection->connect()\\n#1 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/db\\\/connection.php(111): Doctrine\\\\DBAL\\\\Connection->executeQuery('SELECT `configv...', Array, Array, NULL)\\n#2 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/appconfig.php(259): OC\\\\DB\\\\Connection->executeQuery('SELECT `configv...', Array)\\n#3 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/app.php(219): OC\\\\AppConfig->getValues(false, 'enabled')\\n#4 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/app.php(71): OC_App::getEnabledApps()\\n#5 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/base.php(539): OC_App::loadApps(Array)\\n#6 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/base.php(1007): OC::init()\\n#7 \\\/var\\\/www\\\/html\\\/owncloud\\\/index.php(34): require_once('\\\/var\\\/www\\\/html\\\/o...')\\n#8 {main}\",\"File\":\"\\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/db\\\/connection.php\",\"Line\":33}","level":4,"time":"2015-03-06T19:11:02+00:00"}
```
which frankly points to 5 different files and i;m not sure where to look for problems...

**Comment by koumdros** [comment]
back with a solution that worked for me. I had to 
remove old phpm which removed also all owncloud files

then install the owncloud SCl collection ( whci also strangely installed php-common version 5.3.3).

then i had to run the upgrade script from within the /var/www/html/owncloud directory.

as the php version was not visible to the apache user under which the comand had to be run, I had to run it hardcoded.

 sudo -u apache /opt/rh/php54/root/usr/bin/php occ upgrade

this went like a charm, i was able to enter owncloud, and reinstall all the apps, like callendar and dcouments as they are no longer in the core install.

**Comment by ciroiriarte** [comment]
I'm seeing a wrong dependency on php-ldap currently:

[root@d01 ~]# rpm -e php-ldap
error: Error de dependencias:
        php-ldap se necesita para (instalado) owncloud-app-user_ldap-8.0.3-2.1.noarch

The requested package should be php54-php-ldap

**Comment by RealRancor** [comment]
@ciroiriarte 
I think its better to create a new issue instead of reporting this in a closed issue

**Comment by jnweiger** [comment]
@ciroiriarte Please create a new issue for this. Please also add how exactly you installed your system. I assume its CentOS6 with some kind of php upgrade.
